src/java.base/share/classes/java/io/InputStreamReader.java

```diff
@@ -47,8 +47,11 @@
  * BufferedReader.  For example:
  *
  * {@snippet lang=java :
- *     BufferedReader in = new BufferedReader(new InputStreamReader(anInputStream));
+ *     BufferedReader in = new BufferedReader(
+ *         new InputStreamReader(System.in, System.getProperty("stdin.encoding")));
  * }
+ * This example also demonstrates how to wrap {@link System#in} with {@link
+ * System##stdin.encoding stdin.encoding}.
```

Comment:
This example communicate two concerns, one is buffering, the other is the charset to use when running from System.in. I think we should separate these two, make with two examples.

Reply:
Made it two examples.